security/security-core/src/main/java/io/camunda/security/configuration/InitializationConfiguration.java

```diff
@@ -47,4 +53,19 @@ public Map<String, Map<String, Collection<String>>> getDefaultRoles() {
   public void setDefaultRoles(final Map<String, Map<String, Collection<String>>> defaultRoles) {
     this.defaultRoles = defaultRoles;
   }
+
+  public String getIdentifierRegex() {
+    return identifierRegex;
+  }
+
+  public void setIdentifierRegex(final String identifierRegex) {
+    this.identifierRegex = identifierRegex;
+  }
+
+  public Pattern getIdentifierPattern() {
+    if (identifierPattern == null) {
+      identifierPattern = Pattern.compile(identifierRegex);
```

Comment:
❓ What happens if the pattern doesn't compile? Do we fail with a clear message? Could we verify this on startup so we fail early?